pkg/controllers/management/auth/globalroles/globalrolebinding_handler.go

```diff
@@ -585,6 +598,115 @@ func (grb *globalRoleBindingLifecycle) grantRestrictedAdminUserClusterPermission
+	}
+
+	roleBindingUIDs := []types.UID{}
+
+	for ns := range gr.NamespacedRules {
+		_, err := grb.nsLister.Get("", ns)
+		if err != nil {
+			returnError = multierror.Append(returnError, errors.Wrapf(err, "couldn't get namespace %s", ns))
+			continue
+		}
+
+		roleName := gr.Name + "-" + ns
+		roleRef := v1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "Role",
+			Name:     roleName,
+		}
+
+		subject := rbac.GetGRBSubject(globalRoleBinding)
+		rbName := globalRoleBinding.Name + "-" + ns
+		roleBinding, err := grb.roleBindingLister.Get(ns, rbName)
+		if err == nil {
+			if !reflect.DeepEqual(roleRef, roleBinding.RoleRef) || roleBinding.Labels[grbOwnerLabel] != globalRoleBinding.Name {
+				// Since roleRef is immutable, we have to delete and recreate the RB
+				err = grb.roleBindings.DeleteNamespaced(roleBinding.Namespace, roleBinding.Name, &metav1.DeleteOptions{})
+				if err != nil {
+					returnError = multierror.Append(returnError, err)
+					continue
+				}
+			} else {
+				// If the RB is valid, do nothing
+				roleBindingUIDs = append(roleBindingUIDs, roleBinding.UID)
+				continue
+			}
+		} else if !apierrors.IsNotFound(err) {
+			returnError = multierror.Append(returnError, err)
+			continue
+		}
+
+		// Create a newRoleBinding
+		newRoleBinding := &v1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      rbName,
+				Namespace: ns,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion: globalRoleBinding.APIVersion,
+						Kind:       globalRoleBinding.Kind,
+						UID:        globalRoleBinding.UID,
+						Name:       globalRoleBinding.Name,
+					},
+				},
+				Labels: map[string]string{
+					grbOwnerLabel: wrangler.SafeConcatName(globalRoleBinding.Name),
+				},
+			},
+			RoleRef:  roleRef,
+			Subjects: []v1.Subject{subject},
+		}
+
+		createdRB, err := grb.roleBindings.Create(newRoleBinding)
+		if err != nil {
+			returnError = multierror.Append(returnError, err)
+			continue
+		}
+		roleBindingUIDs = append(roleBindingUIDs, createdRB.UID)
+
+	}
+
+	// get all the roleBindings claiming to be owned by this GRB and remove any that shouldn't exist
+	r, err := labels.NewRequirement(grbOwnerLabel, selection.Equals, []string{globalRoleBinding.Name})
+	if err != nil {
+		return multierror.Append(returnError, errors.Wrapf(err, "couldn't create label: %s", grOwnerLabel))
+	}
+
+	rbs, err := grb.roleBindingLister.List("", labels.NewSelector().Add(*r))
+	if err != nil {
+		returnError = multierror.Append(returnError,
+			errors.Wrapf(err, "couldn't list roleBindings with label %s : %s", grbOwnerLabel, globalRoleBinding.Name))
+	} else {
+		err = grb.purgeInvalidNamespacedRBs(rbs, roleBindingUIDs)
+		if err != nil {
+			returnError = multierror.Append(returnError, err)
+		}
+	}
+	return returnError
+}
+
+// purgeInvalidNamespacedRulesRBs removes any roleBindings that aren't in the namespaces listed in the associated GlobalRole.namespacedRules
```

Comment:
```suggestion
// purgeInvalidNamespacedRBs removes any roleBindings that aren't in the namespaces listed in the associated GlobalRole.namespacedRules
```

Adjust comment to be the same as the function name